Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserTableModel.cpp

```diff
@@ -198,6 +202,113 @@ namespace AzToolsFramework
             emit layoutChanged();
             endResetModel();
         }
+        
+        Qt::DropActions AssetBrowserTableModel::supportedDropActions() const
+        {
+            return Qt::CopyAction | Qt::MoveAction;
+        }
+
+        Qt::ItemFlags AssetBrowserTableModel::flags(const QModelIndex& index) const
+        {
+            Qt::ItemFlags defaultFlags = QAbstractItemModel::flags(index) | Qt::ItemIsEditable;
+
+            if (index.isValid())
+            {
+                // We can only drop items onto folders so set flags accordingly
+                const AssetBrowserEntry* item =
+                    mapToSource(index).data(AssetBrowserModel::Roles::EntryRole).value<const AssetBrowserEntry*>();
+                if (item)
+                {
+                    if (item->GetEntryType() == AssetBrowserEntry::AssetEntryType::Product ||
+                        item->GetEntryType() == AssetBrowserEntry::AssetEntryType::Source)
+                    {
+                        return Qt::ItemIsDragEnabled | defaultFlags;
+                    }
+                    if (item->GetEntryType() == AssetBrowserEntry::AssetEntryType::Folder)
+                    {
+                        return Qt::ItemIsDragEnabled | Qt::ItemIsDropEnabled | defaultFlags;
+                    }
+                }
+            }
+            return defaultFlags;
+        }
+
+        bool AssetBrowserTableModel::dropMimeData(
+            const QMimeData* data, Qt::DropAction action, int row, int column, const QModelIndex& parent)
+        {
+            if (action == Qt::IgnoreAction)
+            {
+                return true;
+            }
+
+            auto sourceparent = mapToSource(parent).data(AssetBrowserModel::Roles::EntryRole).value<const AssetBrowserEntry*>();
```

Comment:
Nit: For readability
```suggestion
            auto sourceParent = mapToSource(parent).data(AssetBrowserModel::Roles::EntryRole).value<const AssetBrowserEntry*>();
```

Reply:
Changed.